onnxruntime/core/graph/contrib_ops/quantization_defs.cc

```diff
@@ -1132,88 +1132,225 @@ If mask is provided, mask index (that is position of first 0 in mask, or number
+)DOC";
 
-  ONNX_MS_OPERATOR_SET_SCHEMA(QOrderedLayerNormalization, 1,
-                              OpSchema()
-                                  .SetDoc("QOrderedLayerNormalization")
-                                  .Attr("axis",
-                                        "The first normalization dimension: normalization "
-                                        "will be performed along dimensions axis "
-                                        ": rank(inputs).",
-                                        AttributeProto::INT,
-                                        static_cast<int64_t>(-1))
-                                  .Attr("epsilon", "The epsilon value to use to avoid division by zero.",
-                                        AttributeProto::FLOAT, 1e-5f)
-                                  .Attr("order_X", "cublasLt order of input X. Default is ROW MAJOR.",
-                                        AttributeProto::INT, static_cast<int64_t>(1))
-                                  .Attr("order_Y",
-                                        "cublasLt order of matrix Y, must be same as order_X. "
-                                        "Default is ROW MAJOR.",
-                                        AttributeProto::INT, static_cast<int64_t>(1))
-                                  .AllowUncheckedAttributes()
-                                  .Input(0, "X", "Input data tensor from the previous layer.", "Q")
-                                  .Input(1, "scale_X", "scale of the quantized X", "S")
-                                  .Input(2, "scale", "Scale tensor, i.e., gamma vector.", "F")
-                                  .Input(3, "B", "Bias tensor.", "F", OpSchema::Optional)
-                                  .Input(4, "scale_Y", "scale of the quantized X", "S")
-                                  .Output(0, "Y", "Output data tensor.", "Q")
-                                  .TypeConstraint("F", {"tensor(float16)", "tensor(float)"},
-                                                  "Constrain input gamma and bias could be float16/float tensors. "
-                                                  "float may get better precision, float16 runs faster.")
-                                  .TypeConstraint("S", {"tensor(float)"}, "quantization scale must be float tensors.")
-                                  .TypeConstraint("Q", {"tensor(int8)"}, "quantization tensor must be int8 tensors.")
-                                  .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
-                                    propagateShapeAndTypeFromFirstInput(ctx);
-                                    propagateElemTypeFromInputToOutput(ctx, 0, 0);
-                                  }));
+  ONNX_MS_OPERATOR_SET_SCHEMA(
+      QOrderedMatMul,
+      1,
+      OpSchema()
+          .SetDoc(QOrderedMatMul_ver1_doc)
+          .Attr("order_A", "cublasLt order of matrix A", AttributeProto::INT)
```

Comment:
Do we want to define defaults for the orders (ORDER_ROW) ?

Reply:
All have pros/cons. Let's force it to be there. :)